### PR TITLE
inputs: cache inputresolver results

### PR DESCRIPTION
**Pull request by fho**
```
        inputs: cache inputresolver results

        An app often has multiple tasks that use the same input definitions.  Resolving
        inputs can be time intensive, especially the Gosource resolver.

        To speed up input resolving, cache inputresolver results instead of resolving
        the same input definitions multiple times.

        An LRU cache is used, the max. number of entries is limited to 512 to prevent
        that the memory usage can grow infinite.

-------------------------------------------------------------------------------
        init db: add missing newline to end of printed error message
```

### CODE REVIEW

**Review by FranciscoKurpiel — COMMENTED** [review]
I have a small suggestion on the statistics.

This new cache system will use an unbounded amount of RAM. Maybe there should be a limit, after which we stop caching?

File: pkg/baur/inputresolver_cache.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+func (i *inputResolverCache) HitRatio() float64 {
+	i.mu.Lock()
+	defer i.mu.Unlock()
+
+	if i.hits == 0 && i.miss == 0 {
+		return 0
+	}
+
+	return (float64(i.hits) / float64(i.hits+i.miss)) * 100
+}
```
[the remark this answers is not in the document]
Comment (fho) [reply pkg/baur/inputresolver_cache.go:49]: great idea, thanks

[the remark this answers is not in the document]
Comment (fho) [reply pkg/baur/inputresolver_cache.go]: changed

**Review by FranciscoKurpiel — APPROVED** [review]
Excelente improvement! :100:

**Pull request merged**